app/main/views/frameworks.py

```diff
@@ -1353,27 +1353,38 @@ def sign_framework_agreement(framework_slug):
 
     supplier = data_api_client.get_supplier(current_user.supplier_id)["suppliers"]
     company_details = get_company_details_from_supplier(supplier)
-    form = SignFrameworkAgreementForm()
+    declaration = data_api_client.get_supplier_declaration(current_user.supplier_id, framework_slug).get('declaration')
```

Comment:
A general point about `sign_framework_agreement` - it's now ~100 lines long. Would it be worth breaking bits out into functions to make it easier to understand what's going on?